modules/api/src/main/java/org/apache/ignite/sql/BatchedArguments.java

```diff
@@ -73,8 +72,132 @@ public static BatchedArguments of(List<List<Object>> args) {
      * @return {@code this} for chaining.
      */
     public BatchedArguments add(Object... args) {
-        add(List.of(args));
+        Objects.requireNonNull(args, "args");
+
+        return add(List.of(args));
+    }
+
+    /**
+     * Appends arguments to a batch.
+     *
+     * @param argsList Arguments list.
+     * @return {@code this} for chaining.
+     */
+    public BatchedArguments add(List<Object> argsList) {
+        Objects.requireNonNull(argsList, "argsList");
+
+        if (argsList.isEmpty()) {
+            throwEmptyArgumentsException();
```

Comment:
The user can modify the list after this method call, do we handle this?

Reply:
Thanks, I believe I fixed this.